queuebacca-core/src/main/java/io/axonif/queuebacca/subscribing/Subscriber.java

```diff
@@ -193,23 +236,27 @@ private void check() throws InterruptedException {
 
 		private ThreadPoolWorkExecutor.WorkOrder newWorkOrder(IncomingEnvelope<M> envelope) {
 			return () -> {
-				MessageContext messageContext = new MessageContext(envelope.getMessageId(), envelope.getReadCount(), envelope.getFirstReceived(), envelope.getRawMessage());
-				MDC.put("queuebaccaMessageId", messageContext.getMessageId());
-				MDC.put("queuebaccaMessageReadCount", String.valueOf(messageContext.getReadCount()));
+				MDC.put("queuebaccaMessageId", envelope.getMessageId());
+				MDC.put("queuebaccaMessageReadCount", String.valueOf(envelope.getReadCount()));
+
+				int currentReadCountThreshold = consumerSelector.currentReadCountThreshold(envelope.getReadCount());
+				int currentConsumersReadCount = envelope.getReadCount() - currentReadCountThreshold;
+				MessageContext messageContext = new MessageContext(envelope.getMessageId(), currentConsumersReadCount, envelope.getFirstReceived(), envelope.getRawMessage());
 				try {
 					MessageResponse messageResponse;
 					try {
 						long start = System.currentTimeMillis();
```

Comment:
I think I have ingrained somewhere that for relative timings, using `System.nanoTime()` is preferred. As opposed to being a wall clock that can change (i.e. from leap seconds, OS level date changes) it is a precision timer that will always produce positive elapsed time.

Reply:
Probably worth a ticket then. This was introduced a long time ago.